stan-dev/math
# Adding test for multiple translation units

#### Submisison Checklist


#### Summary:
Adds a test for compiling Stan with multiple translation units.

#### Intended Effect:
This test checks that the math library is able to be compiled under multiple translation units. I verified that copying these files to v2.10.0 will cause a test failure.

#### How to Verify:
```
./runTests.py test/unit/multiple_translation_units_test.cpp
```

If you want to see it fail, copy these files to v2.10.0:
- make/tests
- makefile
- test/unit/multiple_translation_units1.cpp
- test/unit/multiple_translation_units2.cpp
- test/unit/multiple_translation_units_test.cpp
And run the test.

#### Side Effects:
None. Test only.

#### Documentation:
I added a note to the `make/tests` file, but there isn't any additional doc. The test gets run with the unit tests.

#### Reviewer Suggestions: 
Anyone. @rgiordan, want to verify?

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)
